dds/DCPS/transport/tcp/TcpConnection.cpp

```diff
@@ -584,7 +584,7 @@ OpenDDS::DCPS::TcpConnection::active_reconnect_i()
 
     TcpConnection* pconn = this;
     int ret = transport.connector_.connect(pconn, this->remote_address_,  ACE_Synch_Options::asynch);
-    if (ret == -1 && errno != EWOULDBLOCK) {
+    if (ret == -1 && errno != EWOULDBLOCK && errno != EINPROGRESS) {
```

Comment:
Does this also need to change in `TcpTransport::connect_datalink`?

Reply:
I did this change based on the error log from the scoreboard. I don't have windows machine to test it. However, I don't quite understand why the code in TcpTransport::connect_datalink() work as it is while I need to check EINPROGRESS for reconnect.  Maybe it's because I reused the TcpTransport::connector_ object to reconnect and the behavior of the ACE_Connect is not consistent when you use the same connect object to invoke asynchronous connect in the second time.  As this moment, I won't change TcpTransport the same way simply because there is no tests showing errors there.